geo/src/algorithm/coordinate_position.rs

```diff
@@ -2,12 +2,12 @@ use crate::algorithm::{
     bounding_rect::BoundingRect, dimensions::HasDimensions, intersects::Intersects,
 };
 use crate::{
-    Coordinate, GeoNum, Geometry, GeometryCollection, Line, LineString, MultiLineString,
-    MultiPoint, MultiPolygon, Point, Polygon, Rect, Triangle,
+    Coordinate, GeoNum, Geometry, GeometryCollection, GeometryCow, Line, LineString,
+    MultiLineString, MultiPoint, MultiPolygon, Point, Polygon, Rect, Triangle,
 };
 
 /// The position of a `Coordinate` relative to a `Geometry`
-#[derive(PartialEq, Clone, Debug)]
+#[derive(PartialEq, Clone, Copy, Debug)]
 pub enum CoordPos {
```

Comment:
This might be a point of confusion if you are comparing to the reference JTS code...

There are two enums JTS uses ubiquitously in the geomgraph module:

`JTS::Location: { Interior, Boundary, Exterior }` and `JTS::Position: { On, Left, Right }`

Somewhat confusingly `JTS::Location` maps to `geo::CoordPos` and we didn't yet have a concept of `JTS::Position`, and I wanted to call it something other than `Position` to differentiate it from our `CoordPos`, so I've introduced `pub(crate) enum geomgraph::Direction { On, Left, Right }`. 

Unfortunately, this somewhat collides with `pub enum orient::Direction { Default, Reversed }`, but I don't have a better solution at this point. FWIW it's not `pub`, so we can always revisit later without breaking public APIs.